### PR TITLE
Add single node GCP Terraform deploy option

### PR DESCRIPTION
**Pull request by jrhizor**
Adds the ability to deploy a single node to GCP with minimal effort with Terraform.

Remaining todos:

* [ ] Grant the ability for an external IAM to be an `instanceAdmin` and access to Stackdriver metrics for that instance. 
* [ ] Put docs in actual docs, not `README.md`
* [ ] Add deletion protection? (open to comment on if this is useful/desired)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368228097242) by [Unito](https://www.unito.io)

### CODE REVIEW

File: terraform/gcp/single-node/main.tf [hunk 1]
```diff
@@ -0,0 +1,85 @@
+resource "google_compute_network" "airbyte-network" {
```
Comment (michel-tricot) [inline terraform/gcp/single-node/main.tf:1]: I think we should only focus on creating the resources that are related to the instance, not network nor subnetworks, nor NAT. These should be specified as variables so you can hook your environement

File: terraform/gcp/single-node/main.tf [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  name    = "airbyte-nat"
+  network = google_compute_network.airbyte-network.self_link
+}
+
+resource "google_compute_router_nat" "airbyte-nat" {
+  name                               = "airbyte-nat"
+  router                             = google_compute_router.airbyte-router.name
+  nat_ip_allocate_option             = "AUTO_ONLY"
+  source_subnetwork_ip_ranges_to_nat = "ALL_SUBNETWORKS_ALL_IP_RANGES"
+}
```
Comment (michel-tricot) [inline terraform/gcp/single-node/main.tf:33]: I woudn't put that here.

File: terraform/gcp/single-node/main.tf [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  name                    = "airbyte-network"
+  auto_create_subnetworks = "false"
+}
+
+resource "google_compute_subnetwork" "airbyte-subnetwork" {
+  name                     = "airbyte-subnetwork"
+  ip_cidr_range            = "10.128.0.0/9"
+  network                  = google_compute_network.airbyte-network.self_link
+  private_ip_google_access = true
+}
```
Comment (michel-tricot) [inline terraform/gcp/single-node/main.tf:11]: I wouldn't put that here

**Review by michel-tricot — CHANGES_REQUESTED** [review]

**Comment by sherifnada** [comment]
@jrhizor is this blocked on review?

**Pull request closed without merging**